### ISSUE TITLE
Google Log in/Sign Up failing on 0.5

### ISSUE THREAD

**Issue by ApocalypticFish** [toggl/mobileapp#2501]
Attempted to "Sign up with Google" on 0.5, had the prompt for selecting Google account.

Selected a new Google account with no existing Toggl login.

Received error "Sorry, we're having trouble signing you up. Please try again"

On second attempt the loading wheel animation gets stuck on the blue "Sign Up" bar.

**Comment by amulware** [comment]
I suspect we might not have made the changes we made to fix the login to the signup, or something like that.

**Comment by pe0ny** [comment]
@ApocalypticFish Does Google log in work for you?

**Comment by amulware** [comment]
He said internally that it does.

**Comment by ApocalypticFish** [comment]
Just tried again and Google log in is failing.

Not sure how it appeared to work once before.

**Comment by amulware** [comment]
This is now critical and should be fixed on the `boron` release banch.

**Comment by juanlaube** [comment]
Shall we close this issue?

**Comment by amulware** [comment]
What is the current state? Is it still broken on play store builds, but working when installing the same apk manually?

**Comment by juanlaube** [comment]
TL;DR I added the fingerprint of our app signing certificate to the developer console and the problem was solved without a new build.


We are using the new App Signing for Play Store. The signing happens in two steps:
1. We sign the APK with an upload certificate and upload that to Play Store
2. Play Store removes that signing, and resigns with an App Signing certificate (i.e: like the distribution certificate from Apple)

Google Sign In requires us to tell them the fingerprint of the certificate we used to sign the APK, the problem is we told Firebase the fingerprint of our upload certificate, but we didn’t do the same for our App Signing certificate. By adding that fingerprint to the developer console the problem was solved without rebuilding, because the problem was validating the tokens on Google’s side, not ours.

I still have some unanswered questions though:
1. What’s the difference between all the client ids we see in the google-services.json file
2. Why this works with the default client id? Why not the specific client ids for each certificate?
3. And finally… what the fuck, Google‽
